Userland/Libraries/LibWeb/Layout/TextNode.cpp

```diff
@@ -377,15 +388,37 @@ void TextNode::compute_text_for_rendering()
             ++index;
     };
 
+    const CSS::Size max_width = computed_values().width();
+    // FIXME: Support non-px sizes for maximum length determination
```

Comment:
This seems like it would have to work very differently to support layout-dependent sizes like `50%`.

Conceptually speaking, this code runs earlier in the layout pipeline, long before we know the container size. I think this needs to be done differently, at a later stage.